nodejs/koa/auth.js

```diff
@@ -0,0 +1,34 @@
+const passport      = require('koa-passport');
+const LocalStrategy = require('passport-local').Strategy;
+const { poolResolve, odbc } = require('./db');
+
+// We can just pass the username back and forth to serialize/deserialize, other
+// applications may need to add more logic
+passport.serializeUser((user, done) => {
+  done(null, user)
+});
+
+passport.deserializeUser(async (user, done) => {
+  done(null, user)
+});
+
+// When we need to authenticate, use ODBC to create a database connection and
+// make sure the credentials were valid
+passport.use(new LocalStrategy(async (username, password, done) => {
+  try {
+    // connect to the database to check our credentials! The profile will
+    // also be used to handle authority lists for running Db2 queries
+    pool = await odbc.pool(`DSN=*LOCAL;UID=${username};PWD=${password};`);
```

Comment:
Should verify that username is nonzero length and doesn't start with `*` to prevent a malicious user from logging in with `*CURRENT`

Reply:
Fixed in latest push